src/test/init/features/functions.spec.ts

```diff
@@ -0,0 +1,236 @@
+        promptOnceStub.onCall(3).resolves("javascript");
+        promptStub.onFirstCall().callsFake((functions: any): Promise<void> => {
+          functions.lint = true;
+          return Promise.resolve();
+        });
+        promptStub.onSecondCall().resolves(false); // install dependencies?
+        askWriteProjectFileStub = sandbox.stub(config, "askWriteProjectFile");
+        askWriteProjectFileStub.resolves(true);
+
+        const setupExpected = [
+          {
+            source: "testsource",
+            codebase: "testcodebase",
+            ignore: ["node_modules", ".git", "firebase-debug.log", "firebase-debug.*.log"],
+            predeploy: ['npm --prefix "$RESOURCE_DIR" run lint'],
+          },
+          {
+            source: "testsource2",
+            codebase: "testcodebase2",
+            ignore: ["node_modules", ".git", "firebase-debug.log", "firebase-debug.*.log"],
+            predeploy: ['npm --prefix "$RESOURCE_DIR" run lint'],
+          },
+        ];
+        const writeFilesExpected = [
+          `testsource2/package.json`,
+          `testsource2/.eslintrc.js`,
+          `testsource2/index.js`,
+          `testsource2/.gitignore`,
+        ];
+
+        await doSetup(setup, config, options);
+
+        expect(_.get(setup, "config.functions")).to.deep.equal(setupExpected);
+        expect(askWriteProjectFileStub.getCalls().map((call) => call.args[0])).to.deep.equal(
+          writeFilesExpected
+        );
+      });
+
+      it("reinitializes an existing codebase", async () => {
+        const { setup, config } = createExistingTestSetupAndConfig();
+        promptOnceStub.onCall(0).resolves("reinit");
+        promptOnceStub.onCall(1).resolves("testcodebase");
+        promptOnceStub.onCall(2).resolves("javascript");
+        promptStub.onFirstCall().callsFake((functions: any): Promise<void> => {
+          functions.lint = true;
+          return Promise.resolve();
+        });
+        promptStub.onSecondCall().resolves(false);
+        askWriteProjectFileStub = sandbox.stub(config, "askWriteProjectFile");
+        askWriteProjectFileStub.resolves(true);
+
+        const setupExpected = [
+          {
+            source: "testsource",
+            codebase: "testcodebase",
+            ignore: ["node_modules", ".git", "firebase-debug.log", "firebase-debug.*.log"],
+            predeploy: ['npm --prefix "$RESOURCE_DIR" run lint'],
+          },
+        ];
+        const writeFilesExpected = [
+          `testsource/package.json`,
+          `testsource/.eslintrc.js`,
+          `testsource/index.js`,
+          `testsource/.gitignore`,
+        ];
+
+        await doSetup(setup, config, options);
+
+        expect(_.get(setup, "config.functions")).to.deep.equal(setupExpected);
+        expect(askWriteProjectFileStub.getCalls().map((call) => call.args[0])).to.deep.equal(
+          writeFilesExpected
+        );
+      });
+      // TODO: legacy config
```

Comment:
going to do the todo?

Reply:
Ah I think I decided to not do it because the "legacy config" test I had in mind wouldn't actually test anything that hadn't been covered by the previous tests